### DISCUSSION TITLE
Libre, open-source, privacy-friendly hosting for uBO projects

### DISCUSSION THREAD

**Discussion by throwaway-d** [uBlockOrigin/uBlock-issues#2418] [Ideas]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

At first, I sent emails to @gorhill but I only received automatic replies, so I made a throwaway account and a throwaway email for this issue. When this issue is satisfiedly solved, I would delete this account.

### A. Consider mirroring repos to Codeberg
I believe you should mirror your repos to Codeberg because:
1. Codeberg is libre, free, open-source, privacy-friendly, non-profit
2. GitHub is privacy-invasive, proprietary
3. People on Codeberg like me can contribute to the project
4. GitHub can take down your repos

Here are some sources for you:
1. https://sfconservancy.org/GiveUpGitHub/
2. https://github.com/andersju/webbkoll/issues/35
3. https://github.com/ungoogled-software/ungoogled-chromium/discussions/1894
4. https://cadence.moe/blog/2022-07-03-git-forge-opinions-github-gitlab-gitea-sourcehut
5. https://github.com/humanetech-community/awesome-humane-tech/issues/33
6. https://docs.codeberg.org/advanced/migrating-repos/

### A specific URL where the issue occurs.

https://codeberg.org

### Steps to Reproduce

https://weblate.org

### Expected behavior

I expected this will be read by @gorhill. Then we will a have a discussion. And uBlock Origin will be on Codeberg and Weblate

### Actual behavior

I created this issue

### uBO version

1.45.2

### Browser name and version

Edge 108.0.1462.54 (Official build) (64-bit)

### Operating System and version

Windows 10 Pro 21H2 KB5021233

**Comment by uBlock-user** [comment]
> Codeberg is libre, free, open-source, privacy-friendly, non-profit
    GitHub is privacy-invasive, proprietary
    People on Codeberg like me can contribute to the project
    GitHub can take down your repos

Duplicate of https://github.com/uBlockOrigin/uBlock-issues/issues/74

**Reply by throwaway-d** [reply]
> > Codeberg is libre, free, open-source, privacy-friendly, non-profit
> > GitHub is privacy-invasive, proprietary
> > People on Codeberg like me can contribute to the project
> > GitHub can take down your repos
> 
> Duplicate of #74

@uBlock-user Not exactly, entirely though.

**Reply by uBlock-user** [reply]
Doesn't matter, the project is not going to get removed from Github, that is all.

**Reply by throwaway-d** [reply]
> Doesn't matter, the project is not going to get removed from Github, that is all.

So yeah, I used the word _"mirror"_. I don't think it is a good idea to entirely remove from GitHub because there are still lots of contributors on there.

**Reply by uBlock-user** [reply]
Already mirrored at https://gitlab.com/gorhill/uBlock . Despite having mirror/s, the project will still continue to be operated from GitHub for development, so having yet another mirror doesn't solve anything.

**Reply by throwaway-d** [reply]
> Despite having mirror/s, the project will still continue to be operated from GitHub for development, so having yet another mirror doesn't solve anything.

You can create issues, contribute, view the repos on there without going to GitHub!

**Reply by throwaway-d** [reply]
Example:
https://github.com/libredirect/libredirect
https://codeberg.org/LibRedirect/libredirect

**Reply by krystian3w** [reply]
>Already mirrored at https://gitlab.com/gorhill/uBlock .

8 months gap, needed reclick options in gitlab settings if team missed e-mails about stopped automaticaly copy changes.

**Comment by throwaway-d** [comment]
We also talk about Weblate, too. Don't forget.

**Reply by krystian3w** [reply]
I think obvious tracking is blocked by EasyPrivacy.

**Reply by throwaway-d** [reply]
> I think obvious tracking is blocked by EasyPrivacy.

Uhh... Sorry but I don't see how is this related to the issue. Can you explain more, please? Thank you for your discussion!

**Reply by krystian3w** [reply]
>  ### instead of proprietary, tracker-filled Crowdin

**Reply by throwaway-d** [reply]
> > instead of proprietary, tracker-filled Crowdin

okay

**Reply by throwaway-d** [reply]
Also on Crowdin, you cannot even view translations without an account. On the other hand with Weblate, you can view the translations without a account and make suggestions, then a contributor with an account approves them.

Weblate is integrated with Git. So you can also make PRs to translate from a Git service.

**Comment by gorhill** [comment]
There is a lot of friction for moving to another hosting solution:

- Losing bandwidth -- superficially looking, I couldn't find any mention of bandwidth limits
- GitHub Actions -- this is a huge time saver for me
- Transferring issues -- is this possible? Issues here represent a technical knowledge base we can't live without
- Transferring wiki and all edit history -- possible?
- Can issues be submitted by users directly from the popup panel as is currently done in uBO?
- Undergoing such endeavor would essentially stall development for unclear amount of time, and make me spend time on _not-fun_ stuff -- uBO has always and is still a hobby, turning it into a chore is not going to help me motivated to work on it
- Forcing all contributors to modify their current setup and move to another hosting -- we may just lose contributors in the end

There is just too many unknowns including an unknown amount of work for me to want to undergo such change.

**Reply by throwaway-d** [reply]
> There is a lot of friction for moving to another hosting solution:
> * GitHub Actions -- this is a huge time saver for me
> * Forcing all contributors to modify their current setup and move to another hosting -- we may just lose contributors in the end

As I said, we only just _mirror_. The repo will be available and contributable on **both** sites.

**Reply by throwaway-d** [reply]
> * Can issues be submitted by users directly from the popup panel as is currently done in uBO?

I tried testing it but it even didn't work with **GitHub!** It keeps loading indefinitely.
![image](https://user-images.githubusercontent.com/121148372/209134521-81f7dcaa-ba60-4692-b223-ea739f9fadce.png)

**Reply by throwaway-d** [reply]
> * Losing bandwidth -- superficially looking, I couldn't find any mention of bandwidth limits

Oh.. I think you should directly ask https://docs.codeberg.org/contact/ about if they are ready for massive bandwidth rate increase.

**Comment by throwaway-d** [comment]
> Transferring issues -- is this possible? Issues here represent a technical knowledge base we can't live without

Yes, it is possible.
1. Example for issues:

- https://github.com/andersju/webbkoll/issues/35
- https://codeberg.org/dataskydd.net/webbkoll/issues/35
- https://github.com/andersju/webbkoll/issues/33
- https://codeberg.org/dataskydd.net/webbkoll/issues/33
- https://github.com/zyachel/quetre/issues/7
- https://codeberg.org/zyachel/quetre/issues/7

2. Example for commits:
- https://codeberg.org/dataskydd.net/webbkoll/commit/6ddfbc6c71ab92f754252dd4904cf49fb19672a6
- https://github.com/andersju/webbkoll/commit/6ddfbc6c71ab92f754252dd4904cf49fb19672a6

**Reply by krystian3w** [reply]
How mechanism copy github comments and allow recovery delete copy on codeberg after correctly register account.

**Reply by throwaway-d** [reply]
Do you mean how does it migrate comments? It uses Access Token
![image](https://user-images.githubusercontent.com/121148372/209098935-5b16df71-9832-4e2f-8a12-e8911d5e7e66.png)

**Comment by throwaway-d** [comment]
> Transferring wiki and all edit history -- possible?

It is also possible. Example:

- https://github.com/Freeyourgadget/Gadgetbridge/wiki/Home/_history?page=2

- https://codeberg.org/Freeyourgadget/Gadgetbridge/wiki/Home?action=_revision

**Comment by throwaway-d** [comment]
Feel free to ask Codeberg for support: https://docs.codeberg.org/contact/

**Comment by throwaway-d** [comment]
Can the "duplicate" label be removed please?

**Comment by throwaway-d** [comment]
BTW, I made a repo dedicated to mirroring to Codeberg: https://codeberg.org/HexagonCDN/Mirror_to_Codeberg
